text/0000-roadmap-2018.md

```diff
@@ -0,0 +1,412 @@
+
+It’s important to keep in mind two additional factors:
+
+
+- “Shipping” features in this context means they must be *stable*. We may land additional unstable features this year (like `const` generics), but these are separate from the Rust 2018 product. **We will stabilize features individually as they become ready,** ***not*** **in a rush before the epoch release.**
+
+
+- The intent is to ship Rust 2018 in the latter part of the year. The tentative target date is the 1.29 release, which goes into beta on 2018-08-02 and ships on 2018-09-13. That gives us approximately six months to put the product together.
+
+**These two factors together suggest that Rust 2018 will ship largely with** ***language*** **features that are already in nightly in some form today.** Other, faster-moving areas of the product will be developing new material throughout the year.
+
+As always, we will continue to push out new Rust releases on a six week cadence, so a given feature missing the epoch release is by no means fatal. On the other hand, we need to carefully coordinate the work so that the features we *do* ship sit together coherently across the compiler, tools, documentation, libraries and marketing materials.
+
+#### Language
+
+
+> [Rust 2018: Consolidation (](https://deterministic.space/rust-2018.html)@killercup)
+
+The most prominent language work in the pipeline stems from [2017’s ergonomics initiative](https://blog.rust-lang.org/2017/03/02/lang-ergonomics.html). Almost all of the accepted RFCs from the initiative are available on nightly, but polish, testing, and consensus work will take time:
+
+[](https://boats.gitlab.io/blog/post/2018-01-08-goals-for-rust/)
+> [I’d like to reach a final decision to ship or drop all of the ergonomics RFCs that were accepted. I hope to see this completed over the next several mont](https://boats.gitlab.io/blog/post/2018-01-08-goals-for-rust/)[h](https://boats.gitlab.io/blog/post/2018-01-08-goals-for-rust/)[s . . . I hope we can have a clearer (and more spaced out) schedule for this so that their FCPs are staggered.](https://boats.gitlab.io/blog/post/2018-01-08-goals-for-rust/) (@withoutboats)
+
+Among these productivity features are a few “headliners” that will form the backbone of the release:
+
+
+- **Non-lexical lifetimes**. Currently in “alpha” state on nightly, with work ongoing.
+- `**impl Trait**`. Nearing readiness for stabilization FCP.
+- **Generators**. “Beta” state on nightly, but some design issues need resolution.
+- **Module system changes**. Largely usable on nightly; will need testing, feedback, and bikeshedding.
+
+In addition, there are some other headlining features which are nearing stabilization and should ship prior to the epoch:
+
+
+- **SIMD**. The core SIMD intrinsics are nearing readiness for stabilization, and with luck we may be able to stabilize some vendor-agnostics primitives as well.
+- **Custom allocators**. TODO
```

Comment:
Was this to be discussed during RFC or was it intended to be fleshed out before hand and discussed?